### ISSUE TITLE
convert call fails with TypeError in python3?

### ISSUE THREAD

**Issue by jasedit** [vog/texcaller#3]
I'm attempting to read in a LaTeX file with the following code snippet:

```
import texcaller

data = open('paper.tex').read()

pdf, info = texcaller.convert(data, 'LaTeX', 'PDF', 5)
```

And getting the following error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/texcaller-0-py3.5-macosx-10.6-intel.egg/texcaller.py", line 107, in convert
    val = _texcaller.convert(*args)
TypeError: in method 'convert', argument 3 of type 'std::string const &'
```

It looks like argument 3 should be a string, as it is passed in, but I'm not sure if I'm missing something subtle. This is running on OSX El Capitan, using python 3.5 if that's helpful.

**Comment by vog** [comment]
Indeed, Texcaller needs some upgrades to later Python versions.

Unfortunately, I don't know when I'll get around to do that.

Any patches or pull requests are welcome!

**Comment by jasedit** [comment]
I dug into it a bit more, and it looks like it's an issue with Python 3 because encoding to UTF-8 changes the argument types to `bytes`, which isn't the same as `str` as in Python 2. I'm not familiar enough to present an immediate fix, but that at least provides some insight.

**Comment by vog** [comment]
@VictorCarlquist Thanks for pull request #4.

Unfortunately, that pull request fixed "only" an unrelated SWIG issue, not our actual Python 3 issue.

However, it put me on the right track on how to fix the Python 3 compatibility issue, which I finally fixed in commit fe3143d94de9f74ada1faa501bbde828248be8b5.

I fixed some some more issues and released a new version of Texcaller. See also: https://vog.github.io/texcaller/download.html

@jasedit Please check if that works for you.
